# Flipkart data Analysis page 3

In [1]:
import numpy as np
import pandas as pd

In [119]:
df = pd.read_csv('flipkart_spec_cat_brand.csv')

In [120]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19441 entries, 0 to 19440
Data columns (total 12 columns):
Unnamed: 0                 19441 non-null int64
Unnamed: 0.1               19441 non-null int64
product_name               19441 non-null object
product_category_tree      19441 non-null object
retail_price               19441 non-null float64
discounted_price           19441 non-null float64
is_FK_Advantage_product    19441 non-null bool
product_rating             19441 non-null object
overall_rating             19441 non-null object
Brand_names                19441 non-null object
product_category           19441 non-null object
product_specs_refined      19441 non-null object
dtypes: bool(1), float64(2), int64(2), object(7)
memory usage: 1.1+ MB


In [121]:
x = list(df.columns)
x[0],x[1] = 'a','b'
df.columns = x
df.drop(['a','b'],axis=1,inplace=True)
#'product_category_tree','product_name','product_specs_refined','is_FK_Advantage_product'],axis=1,inplace=True)

In [122]:
df.head(2)

product_name  \
0  Alisha Solid Women's Cycling Shorts   
1  FabHomeDecor Fabric Double Sofa Bed   

                               product_category_tree  retail_price  \
0  ["Clothing >> Women's Clothing >> Lingerie, Sl...         999.0   
1  ["Furniture >> Living Room Furniture >> Sofa B...       32157.0   

   discounted_price  is_FK_Advantage_product       product_rating  \
0             379.0                    False  No rating available   
1           22646.0                    False  No rating available   

        overall_rating   Brand_names product_category  \
0  No rating available        Alisha        Clothing    
1  No rating available  FabHomeDecor       Furniture    

                               product_specs_refined  
0  {"product_specification"=>[{"key"=>"Number of ...  
1  {"product_specification"=>[{"key"=>"Installati...

In [123]:
df[df['retail_price']==0].head(3)

product_name  \
12             Sicons All Purpose Tea Tree Dog Shampoo   
70                 Eurospa Cotton Terry Face Towel Set   
707  Techware Microwavable Tea Cups WF13115 - Purpl...   

                                 product_category_tree  retail_price  \
12   ["Pet Supplies >> Grooming >> Skin & Coat Care...           0.0   
70   ["Baby Care >> Baby Bath & Skin >> Baby Bath T...           0.0   
707  ["Kitchen & Dining >> Dinnerware & Crockery >>...           0.0   

     discounted_price  is_FK_Advantage_product       product_rating  \
12                0.0                    False  No rating available   
70                0.0                    False  No rating available   
707               0.0                    False  No rating available   

          overall_rating Brand_names   product_category  \
12   No rating available      Sicons      Pet Supplies    
70   No rating available     Eurospa         Baby Care    
707  No rating available    Techware  Kitchen & Dining    

                                 product_specs_refined  
12   {"product_specification"=>[{"key"=>"Pet Type",...  
70   {"product_specification"=>[{"key"=>"Material",...  
707  {"product_specification"=>[{"key"=>"Brand", "v...

In [124]:
df[df['discounted_price']==0].head(3)

product_name  \
12             Sicons All Purpose Tea Tree Dog Shampoo   
70                 Eurospa Cotton Terry Face Towel Set   
707  Techware Microwavable Tea Cups WF13115 - Purpl...   

                                 product_category_tree  retail_price  \
12   ["Pet Supplies >> Grooming >> Skin & Coat Care...           0.0   
70   ["Baby Care >> Baby Bath & Skin >> Baby Bath T...           0.0   
707  ["Kitchen & Dining >> Dinnerware & Crockery >>...           0.0   

     discounted_price  is_FK_Advantage_product       product_rating  \
12                0.0                    False  No rating available   
70                0.0                    False  No rating available   
707               0.0                    False  No rating available   

          overall_rating Brand_names   product_category  \
12   No rating available      Sicons      Pet Supplies    
70   No rating available     Eurospa         Baby Care    
707  No rating available    Techware  Kitchen & Dining    

                                 product_specs_refined  
12   {"product_specification"=>[{"key"=>"Pet Type",...  
70   {"product_specification"=>[{"key"=>"Material",...  
707  {"product_specification"=>[{"key"=>"Brand", "v...

## The prices which are equal to zero are taken out

In [125]:
df[df['retail_price']==0].info()
df[df['discounted_price']==0].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 78 entries, 12 to 19438
Data columns (total 10 columns):
product_name               78 non-null object
product_category_tree      78 non-null object
retail_price               78 non-null float64
discounted_price           78 non-null float64
is_FK_Advantage_product    78 non-null bool
product_rating             78 non-null object
overall_rating             78 non-null object
Brand_names                78 non-null object
product_category           78 non-null object
product_specs_refined      78 non-null object
dtypes: bool(1), float64(2), object(7)
memory usage: 4.0+ KB
<class 'pandas.core.frame.DataFrame'>
Int64Index: 78 entries, 12 to 19438
Data columns (total 10 columns):
product_name               78 non-null object
product_category_tree      78 non-null object
retail_price               78 non-null float64
discounted_price           78 non-null float64
is_FK_Advantage_product    78 non-null bool
product_rating             78 non-n

In [126]:
df_no_retail_disc = df[df['retail_price']==0]

In [127]:
df_retail_disc = df[df['retail_price']!=0]

## Mean had too much potential of tampering with the inferences with mean data almost equal 3300 for retail, hence filling it with median

In [128]:
df_retail_disc['retail_price'].median()

1049.0

In [129]:
df_retail_disc['discounted_price'].median()

554.0

In [130]:
df_no_retail_disc['retail_price'] = df_retail_disc['retail_price'].median()
df_no_retail_disc['discounted_price'] = df_retail_disc['discounted_price'].median()

C:\Users\h_asp\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':
C:\Users\h_asp\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app


In [131]:
df_no_retail_disc.head(3)

product_name  \
12             Sicons All Purpose Tea Tree Dog Shampoo   
70                 Eurospa Cotton Terry Face Towel Set   
707  Techware Microwavable Tea Cups WF13115 - Purpl...   

                                 product_category_tree  retail_price  \
12   ["Pet Supplies >> Grooming >> Skin & Coat Care...        1049.0   
70   ["Baby Care >> Baby Bath & Skin >> Baby Bath T...        1049.0   
707  ["Kitchen & Dining >> Dinnerware & Crockery >>...        1049.0   

     discounted_price  is_FK_Advantage_product       product_rating  \
12              554.0                    False  No rating available   
70              554.0                    False  No rating available   
707             554.0                    False  No rating available   

          overall_rating Brand_names   product_category  \
12   No rating available      Sicons      Pet Supplies    
70   No rating available     Eurospa         Baby Care    
707  No rating available    Techware  Kitchen & Dining    

                                 product_specs_refined  
12   {"product_specification"=>[{"key"=>"Pet Type",...  
70   {"product_specification"=>[{"key"=>"Material",...  
707  {"product_specification"=>[{"key"=>"Brand", "v...

In [132]:
frames = [df_retail_disc,df_no_retail_disc]

In [133]:
df = pd.concat(frames)

In [134]:
df.head()

product_name  \
0    Alisha Solid Women's Cycling Shorts   
1    FabHomeDecor Fabric Double Sofa Bed   
2                             AW Bellies   
3    Alisha Solid Women's Cycling Shorts   
4  Sicons All Purpose Arnica Dog Shampoo   

                               product_category_tree  retail_price  \
0  ["Clothing >> Women's Clothing >> Lingerie, Sl...         999.0   
1  ["Furniture >> Living Room Furniture >> Sofa B...       32157.0   
2  ["Footwear >> Women's Footwear >> Ballerinas >...         999.0   
3  ["Clothing >> Women's Clothing >> Lingerie, Sl...         699.0   
4  ["Pet Supplies >> Grooming >> Skin & Coat Care...         220.0   

   discounted_price  is_FK_Advantage_product       product_rating  \
0             379.0                    False  No rating available   
1           22646.0                    False  No rating available   
2             499.0                    False  No rating available   
3             267.0                    False  No rating available   
4             210.0                    False  No rating available   

        overall_rating   Brand_names product_category  \
0  No rating available        Alisha        Clothing    
1  No rating available  FabHomeDecor       Furniture    
2  No rating available            AW        Footwear    
3  No rating available        Alisha        Clothing    
4  No rating available        Sicons    Pet Supplies    

                               product_specs_refined  
0  {"product_specification"=>[{"key"=>"Number of ...  
1  {"product_specification"=>[{"key"=>"Installati...  
2  {"product_specification"=>[{"key"=>"Ideal For"...  
3  {"product_specification"=>[{"key"=>"Number of ...  
4  {"product_specification"=>[{"key"=>"Pet Type",...

## Checking for 'No ratings available' in overall_rating and product_rating

In [135]:
df[df['overall_rating'] == 'No rating available'].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 17595 entries, 0 to 19438
Data columns (total 10 columns):
product_name               17595 non-null object
product_category_tree      17595 non-null object
retail_price               17595 non-null float64
discounted_price           17595 non-null float64
is_FK_Advantage_product    17595 non-null bool
product_rating             17595 non-null object
overall_rating             17595 non-null object
Brand_names                17595 non-null object
product_category           17595 non-null object
product_specs_refined      17595 non-null object
dtypes: bool(1), float64(2), object(7)
memory usage: 910.7+ KB


### Seperating and filling it with random integers between 0 to 5, since predicting 17595 on the basis of 1500 data would not be fair

In [136]:
df_no_ratings = df[df['overall_rating'] == 'No rating available']

In [137]:
df_ratings = df[df['overall_rating'] != 'No rating available']

In [138]:
np.random.seed(123)
df_no_ratings['product_rating'] = np.random.randint(0,6,17595)
df_no_ratings['overall_rating'] = np.random.randint(0,6,17595)

C:\Users\h_asp\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
C:\Users\h_asp\Anaconda3\lib\site-packages\ipykernel\__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  app.launch_new_instance()


In [139]:
df_no_ratings.head(6)

product_name  \
0                Alisha Solid Women's Cycling Shorts   
1                FabHomeDecor Fabric Double Sofa Bed   
2                                         AW Bellies   
3                Alisha Solid Women's Cycling Shorts   
4              Sicons All Purpose Arnica Dog Shampoo   
5  Eternal Gandhi Super Series Crystal Paper Weig...   

                               product_category_tree  retail_price  \
0  ["Clothing >> Women's Clothing >> Lingerie, Sl...         999.0   
1  ["Furniture >> Living Room Furniture >> Sofa B...       32157.0   
2  ["Footwear >> Women's Footwear >> Ballerinas >...         999.0   
3  ["Clothing >> Women's Clothing >> Lingerie, Sl...         699.0   
4  ["Pet Supplies >> Grooming >> Skin & Coat Care...         220.0   
5  ["Eternal Gandhi Super Series Crystal Paper We...         430.0   

   discounted_price  is_FK_Advantage_product  product_rating  overall_rating  \
0             379.0                    False               5               3   
1           22646.0                    False               2               5   
2             499.0                    False               4               3   
3             267.0                    False               2               4   
4             210.0                    False               1               3   
5             430.0                    False               3               4   

      Brand_names                                   product_category  \
0          Alisha                                          Clothing    
1    FabHomeDecor                                         Furniture    
2              AW                                          Footwear    
3          Alisha                                          Clothing    
4          Sicons                                      Pet Supplies    
5  Eternal Gandhi  Eternal Gandhi Super Series Crystal Paper Weig...   

                               product_specs_refined  
0  {"product_specification"=>[{"key"=>"Number of ...  
1  {"product_specification"=>[{"key"=>"Installati...  
2  {"product_specification"=>[{"key"=>"Ideal For"...  
3  {"product_specification"=>[{"key"=>"Number of ...  
4  {"product_specification"=>[{"key"=>"Pet Type",...  
5  {"product_specification"=>[{"key"=>"Model Name...

In [140]:
x = np.random.randint(4,5,5)

In [141]:
x

array([4, 4, 4, 4, 4])

In [142]:
frames = [df_no_ratings,df_ratings]

In [143]:
df = pd.concat(frames)
df.head(2)

product_name  \
0  Alisha Solid Women's Cycling Shorts   
1  FabHomeDecor Fabric Double Sofa Bed   

                               product_category_tree  retail_price  \
0  ["Clothing >> Women's Clothing >> Lingerie, Sl...         999.0   
1  ["Furniture >> Living Room Furniture >> Sofa B...       32157.0   

   discounted_price  is_FK_Advantage_product product_rating overall_rating  \
0             379.0                    False              5              3   
1           22646.0                    False              2              5   

    Brand_names product_category  \
0        Alisha        Clothing    
1  FabHomeDecor       Furniture    

                               product_specs_refined  
0  {"product_specification"=>[{"key"=>"Number of ...  
1  {"product_specification"=>[{"key"=>"Installati...

In [147]:
df.to_csv('flipkart_final_cleared.csv',encoding='utf-8')

In [148]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 19441 entries, 0 to 19179
Data columns (total 10 columns):
product_name               19441 non-null object
product_category_tree      19441 non-null object
retail_price               19441 non-null float64
discounted_price           19441 non-null float64
is_FK_Advantage_product    19441 non-null bool
product_rating             19441 non-null object
overall_rating             19441 non-null object
Brand_names                19441 non-null object
product_category           19441 non-null object
product_specs_refined      19441 non-null object
dtypes: bool(1), float64(2), object(7)
memory usage: 1006.2+ KB


## This dataframe is the cleaned data finally and we have now 19441 data with all almost correct data